vowpalwabbit/global_data.h

```diff
@@ -15,6 +15,16 @@ license as described in the file LICENSE.
 #include <inttypes.h>
 #include <climits>
 
+#ifdef _M_CEE
+#pragma managed(push,off)
+#	undef _M_CEE 

```

Comment:
Consider adding a comment to give context for undefing and re-defing _M_CEE around including <thread>. I am curious what issues it causes otherwise.

Reply:
It's possible that the undef and def is redundant. Maybe just the managed bit is required?

Reply:
Not sure. If all you care about is ensuring that /clr does not attempt to compile any of it as managed, per the docs, pushing managed to off should be sufficient. I do not know whether MSVC's <thread> does anything odd with _M_CEE

Reply:
It seems like the _M_CEE  is required